functional-tests/src/test/java/org/infinispan/online/service/endpoint/RESTTester.java

```diff
@@ -29,12 +39,78 @@ public void testIfEndpointIsProtected(URL urlToService) {
       post(urlToService.toString() + "rest/default/should_default_cache_be_accessible_via_REST", "test", 401);
    }
 
+   public void putGetTest(URL urlToService) throws IOException {
+      String value = "value";
+      String key = "restKey";
+
+      String stringUrl = urlToService.toString();
+
+      putMethodPost(stringUrl, key, value);
+      String restValue = getMethod(stringUrl, key);
+      assertEquals(value, restValue);
+
+      String newValue = "newValue";
+      // putting through POST method should return conflict code and not change the value
+      putMethodPost(stringUrl, key, newValue, 409);
+      restValue = getMethod(stringUrl, key);
+      assertEquals(value, restValue);
+
+      // putting through PUT method should change the value and return code 200
+      putMethodPut(stringUrl, key, newValue);
+      restValue = getMethod(stringUrl, key);
+      assertEquals(newValue, restValue);
+
+      removeMethod(stringUrl, key);
+      getMethod(stringUrl, key, 404);
+   }
+
    protected void post(String url, String body, int expectedCode) {
-      given()
-         .body(body)
-      .when()
-         .post(url)
-      .then()
-         .statusCode(expectedCode);
+      given().body(body).when().post(url).then().statusCode(expectedCode);
+   }
+
+   protected void put(String url, String body, int expectedCode) {
+      given().body(body).when().put(url).then().statusCode(expectedCode);
+   }
+
+   protected void delete(String url, String body, int expectedCode) {
+      given().body(body).when().delete(url).then().statusCode(expectedCode);
+   }
+
+   protected String get(String url, String body, int expectedCode) {
+      Response response = given().body(body).when().get(url);
+      response.then().statusCode(expectedCode);
+      return response.andReturn().body().asString();
+   }
+
+   public void putMethodPost(String urlServerAddress, String key, String value, int expectedCode) {
+      post(urlServerAddress.toString() + "rest/" + cache + "/" + key, value, expectedCode);
+   }
+
+   public void putMethodPost(String urlServerAddress, String key, String value) {
+      putMethodPost(urlServerAddress, key, value, 200);
+   }
+
+   public void putMethodPut(String urlServerAddress, String key, String value, int expectedCode) {
+      put(urlServerAddress.toString() + "rest/" + cache + "/" + key, value, expectedCode);
+   }
+
+   public void putMethodPut(String urlServerAddress, String key, String value) {
+      putMethodPut(urlServerAddress, key, value, 200);
+   }
+
+   public String getMethod(String urlServerAddress, String key, int expectedCode) {
```

Comment:
Again, this is something we heavily test in Infinispan. Do we really need to double check here?